packages/common/src/util.ts

```diff
@@ -258,3 +258,43 @@ export function camelize(value: string): string {
       return match.toLowerCase();
     });
 }
+
+/**
+ * Determines if a value is unique in an array
+ * Allows for code like:
+ *
+ * ```js
+ * const ary = [ 1, 2, 3, 3, 4, 5, 1 ];
+ *
+ * const result = ary.filter(unique);
+ *
+ * result => [ 1, 2, 3, 4, 5 ]
+ * ```
+ * @param value - the value to search for
+ * @param index - the index of the searched value
+ * @param ary - the array to search
+ * @returns {boolean} - indicating if the value is unique in the array
+ */
+export function unique(value: any, index: number, ary: any[]): boolean {
+  return ary.indexOf(value) === index;
+}
+
+/**
+ * Concatenates two arrays
+ * Allows for code like:
+ *
+ * ```js
+ * // flatten an array
+ * const ary = [ [ 1, 2, 3 ], [ 3, 4, 5] , [ 1 ] ];
+ *
+ * const result = ary.reduce(concat);
+ *
+ * result => [ 1, 2, 3, 3, 4, 5, 1 ]
+ * ```
+ * @param ary1 - an array to concatenate
+ * @param ary2 - an array to concatenate
+ * @returns {any[]} - a concatenated array
+ */
+export function concat(ary1: any[], ary2: any[]): any[] {
```

Comment:
I think we should ditch this for https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat

Reply:
Well it's not quite the same. the reason i want a concat function with that signature (`concat(ary1, ary2)`)
is so i can have a 2 dimensional array and call `my2dAry.reduce(concat, [])` to flatten it.

Reply:
maybe we should have a `flattenArray()` fn instead?